Consul/Consul.csproj

```diff
@@ -44,7 +45,21 @@
     <DefineConstants>$(DefineConstants);CORECLR</DefineConstants>
   </PropertyGroup>
 
-  <ItemGroup>
-    <DotNetCliToolReference Include="ILRepack" Version="2.0.18" />
-  </ItemGroup>
+  <Target Name="ILMerge" AfterTargets="AfterBuild" Condition=" '$(TargetFramework)' == 'net461' ">
+    <ItemGroup>
+      <BuildArtifacts Include="$(OutputPath)\$(AssemblyName).*" />
+      <ArtifactsToMerge Include="$(OutputPath)\Newtonsoft.Json.dll" />
+    </ItemGroup>
+    <PropertyGroup>
+      <IlmergeCommand>$(ILMergeConsolePath) /out:$(OutputPath)\ILMerge\$(AssemblyName).dll</IlmergeCommand>
+      <IlmergeCommand>$(IlmergeCommand) /keyfile:../assets/consuldotnet.snk</IlmergeCommand>
+      <IlmergeCommand>$(IlmergeCommand) $(OutputPath)\$(AssemblyName).dll @(ArtifactsToMerge->'%(RelativeDir)%(filename)%(extension)', ' ')</IlmergeCommand>
+    </PropertyGroup>
+    <MakeDir Directories="$(OutputPath)\ILMerge" />
+    <Exec Command="$(IlmergeCommand)" ContinueOnError="false" />
+    <Move SourceFiles="%(BuildArtifacts.RelativeDir)\ILMerge\%(filename)%(extension)"
+          DestinationFiles="%(BuildArtifacts.RelativeDir)%(filename)%(extension)"
+          OverwriteReadOnlyFiles="true"/>
+  </Target>
```

Comment:
nitpick: we should delete the `ILMerge` directory and the `@(ArtifactsToMerge)` after the `Move` task

Reply:
Done